dawn/src/dawn/IIR/Extents.h

```diff
@@ -223,8 +222,11 @@ class HorizontalExtent {
   }
   void merge(ast::HorizontalOffset const& other) { merge(HorizontalExtent{other}); }
   bool isPointwise() const { return !impl_ || impl_->isPointwise(); }
-  HorizontalExtent limit(HorizontalExtent const& other) const {
-    return impl_ ? impl_->limit(*other.impl_) : *this;
+  void limit(HorizontalExtent const& other) {
+    if(impl_ && other.impl_)
+      impl_->limit(*other.impl_);
+    else if(other.impl_)
+      *this = other;
```

Comment:
Why did the behavior change here? Is this correct? This line will not limit but rather enlarge.